findfiles.go

```diff
@@ -0,0 +1,53 @@
+package pusher
+
+// This function has been placed in its own file because, in the fullness of
+// time, we expect Pusher to become a library used by the inotify exporter, and
+// for this function to therefore become unused.
+
+import (
+	"log"
+	"os"
+	"path/filepath"
+	"sort"
+	"time"
+)
+
+// FindFiles recursively searches through a given directory to find all the files which are old enough to be eligible for upload.
+func FindFiles(directory string, minFileAge time.Duration) (error, []*LocalDataFile) {
+	eligibleFiles := make([]*LocalDataFile, 0)
+	eligibleTime := time.Now().Add(-minFileAge)
+	totalEligibleSize := int64(0)
+
+	err := filepath.Walk(directory, func(path string, info os.FileInfo, err error) error {
```

Comment:
Add a comment about the efficiency issue noted in the library documentation.  Would need to benchmark this, especially if we roll it out on old kernel.

Reply:
They are worried about the overhead of `sort` on the filenames. I am not and I don't think we should be.